### ISSUE TITLE
Bug: not detecting `SIM105` when loop is present

### ISSUE THREAD

**Issue by jamesbraza** [astral-sh/ruff#10495]
With `ruff==0.3.3`, I think [SIM105](https://docs.astral.sh/ruff/rules/suppressible-exception/) should be triggered here, but it's not:

```python
try:
    for _ in range(1):
        pass
except (NotImplementedError, ValueError):
    pass
```

In other words, I think the above code snippet should be autofixed to:

```python
import contextlib


with contextlib.suppress(NotImplementedError, ValueError):
    for _ in range(1):
        pass
```

**Comment by charliermarsh** [comment]
I think this can be merged into https://github.com/astral-sh/ruff/issues/8593.
